ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -105,9 +110,34 @@ private void setupListeners() {
         .subscribe(EthPV62.NEW_BLOCK_HASHES, this::handleNewBlockHashesFromNetwork);
   }
 
+  @VisibleForTesting
+  void broadcastBlock(final Block block) {
+    blockBroadcaster = new BlockBroadcaster(ethContext);
+    final ProtocolSpec<C> protocolSpec =
+        protocolSchedule.getByBlockNumber(block.getHeader().getNumber());
+    final BlockHeaderValidator<C> blockHeaderValidator = protocolSpec.getBlockHeaderValidator();
+    final BlockHeader parent =
+        protocolContext
+            .getBlockchain()
+            .getBlockHeader(block.getHeader().getParentHash())
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        "Incapable of retrieving header from non-existent parent of "
+                            + block.getHeader().getNumber()
+                            + "."));
+    if (blockHeaderValidator.validateHeader(
+        block.getHeader(), parent, protocolContext, HeaderValidationMode.FULL)) {
+      final UInt256 totalDifficulty =
+          parent.getDifficulty().plus(block.getHeader().getDifficulty());
+      blockBroadcaster.propagate(block, totalDifficulty);
+    }
+  }
+
   private void onBlockAdded(final BlockAddedEvent blockAddedEvent, final Blockchain blockchain) {
     // Check to see if any of our pending blocks are now ready for import
     final Block newBlock = blockAddedEvent.getBlock();
+    broadcastBlock(newBlock);
```

Comment:
This isn't the right place to trigger block propagation.  It will be called after we've processed transactions and imported to the chain - we want to do it before then, so in `importOrSavePendingBlock` just before the `PersistBlockTask`.